src/lightning_lite/utilities/imports.py

```diff
@@ -0,0 +1,164 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""General utilities."""
+import importlib
+import operator
+import platform
+import sys
+from importlib.util import find_spec
+from typing import Callable
+
+import pkg_resources
+from packaging.version import Version
+from pkg_resources import DistributionNotFound
+
+
+def _package_available(package_name: str) -> bool:
+    """Check if a package is available in your environment.
+
+    >>> _package_available('os')
+    True
+    >>> _package_available('bla')
+    False
+    """
+    try:
+        return find_spec(package_name) is not None
+    except ModuleNotFoundError:
+        return False
+
+
+def _module_available(module_path: str) -> bool:
```

Comment:
should we introduce a general minimal helper  package for stuff like this which might also be useful for apps so that apps doesn't have to depend on training?

Reply:
Will redirect this question to my good colleagues @Borda and @carmocca who are better with these kind of discussions :))

Reply:
I think yes. I remember seeing some utilities being copied within apps

Reply:
Yes. I wanted to get this done before this PR because it would simplify a lot.

It's blocked on @Bordas decision in https://github.com/Lightning-AI/devtools/issues/19 and publishing it to PyPI

I have a WIP of the integration in https://github.com/Lightning-AI/lightning/pull/14475

And the devtools issues page contains examples of "tools" to move

Reply:
Great! Once the tools exist, we can simply delete the functions from here and pl. Resolving this conversation then!

Reply:
merged :rabbit: